src/towncrier/test/test_check.py

```diff
@@ -190,51 +196,68 @@ def test_none_stdout_encoding_works(self):
-            # Arrange
-            write("towncrier.toml", "[tool.towncrier]")
-            write("newsfragments/123.feature", "Foo the bar")
-            initial_commit()
+            create_project()
+            # Before any release, the NEWS file might no exist.
+            self.assertNotIn('NEWS.rst', os.listdir())
 
-            call(["git", "checkout", "-b", "next-resease"])
-            call(["towncrier", "--yes", "--version", "1.0"])
+            call(["towncrier", "build", "--yes", "--version", "1.0"])
             commit("Prepare a release")
+            # When missing,
+            # the news file is automatically created with a new release.
+            self.assertIn('NEWS.rst', os.listdir())
 
             # Act
-            result = runner.invoke(_main, ["--compare-with", "main"])
+            result = runner.invoke(towncrier_check, ["--compare-with", "main"])
 
             # Assert
             self.assertEqual(0, result.exit_code, (result, result.output))
             self.assertIn("Checks SKIPPED: news file changes detected", result.output)
 
-    def test_second_release(self):
-        """The checks should be skipped on a branch that modifies the news file."""
+    def test_release_branch(self):
+        """
+        The checks for missing news fragments are skipped on a branch that
+        modifies the news file.
+        This is a hint that we are on a release branch
+        and at release time is expected no not have news-fragment files.
+        """
         runner = CliRunner()
 
         with runner.isolated_filesystem():
-            # Arrange
-            write("towncrier.toml", "[tool.towncrier]")
-            write("newsfragments/123.feature", "Foo the bar")
-            initial_commit()
+            create_project()
 
-            call(["towncrier", "--yes", "--version", "1.0"])
+            # Do a first release without any checks.
+            # And merge the release branch back into the main branch.
+            call(["towncrier", "build",  "--yes", "--version", "1.0"])
             commit("First release")
-
-            write("newsfragments/123.feature", "Foo the bar")
-            commit("Foo the bar")
-
-            call(["git", "checkout", "-b", "next-resease"])
-            call(["towncrier", "--yes", "--version", "2.0"])
+            # The news file is now created.
+            self.assertIn('NEWS.rst', os.listdir())
+            call(["git", "checkout", "main"])
+            call(["git", "merge", "otherbranch", "-m", "Sync release in main branch."])
+
+            # We have a new feature branch that has a news fragment that
+            # will be merged to the main branch.
+            call(["git", "checkout", "-b", "new-feature-branch"])
+            write("foo/newsfragments/456.feature", "Foo the bar")
+            commit("A feature in the second release.")
+            call(["git", "checkout", "main"])
+            call(["git", "merge", "new-feature-branch", "-m", "Sync release in main branch."])
```

Comment:
> "Sync release in main branch."

should probable say "Merge new-feature-branch" or "Merge: Foo the bar" because we are merging a feature branch rather than a release branch